CRM/Contact/BAO/Contact.php

```diff
@@ -2529,8 +2529,10 @@ public static function getValues(&$params, &$values) {
       $values['preferred_communication_method'] = $preffComm;
       $values['preferred_communication_method_display'] = CRM_Utils_Array::value('preferred_communication_method_display', $temp);
 
-      $preferredMailingFormat = CRM_Core_SelectValues::pmf();
-      $values['preferred_mail_format'] = $preferredMailingFormat[$contact->preferred_mail_format];
+      if ($contact->preferred_mail_format) {
+        $preferredMailingFormat = CRM_Core_SelectValues::pmf();
+        $values['preferred_mail_format'] = $preferredMailingFormat[$contact->preferred_mail_format];
+      }
```

Comment:
Fix for https://test.civicrm.org/job/CiviCRM-WebTest-Matrix/CIVIVER=master,label=test-ubu1204-1/lastCompletedBuild/testReport/(root)/WebTest_Contribute_AddPricesetTest/testContributeOfflineforSoftcreditwithApi/ failing due to undefined index error